### PR TITLE
✨ (iterations) Add ability to select iterations 

### PR DESCRIPTION
**Pull request by mattvv**
Adds a navbar item to be able to select which iteration to view for grumpy Alexis

### CODE REVIEW

File: pages/projects/[id].tsx [hunk 1]
```diff
@@ -28,12 +29,22 @@ const Projects = (): JSX.Element => {
   const [filters, setFilters] = useState<Filters>({});
   const stories = useSelector((state: State): Record<string, Story[]> => filterStories(state, id, filters));
 
-  const addFilter = (name: string, filter: Owner | Label): void => {
+  const addFilter = (name: string, filter: Owner | Label | Iteration): void => {
+    if (name === 'iterations') {
+      // @ts-ignore: we know iteration can only be Iteration type
```
Comment (mattvv) [inline pages/projects/[id].tsx:34]: does anyone know how I can imply this using typescript?

Comment (jansen0825) [reply pages/projects/[id].tsx:34]: Not beautiful, but this is how I worked around it. 
If you like this approach better, you can leave a comment on the interface. 

Screen Shot 2020-09-01 at 3 34 08 PM

File: components/Iterations.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+interface IterationsParams {
+  id: string;
+  selectedIteration?: Iteration;
+  addIteration: (name: string, filter: Iteration) => void;
+  removeIteration: (name: string, filter: Iteration) => void;
+}
+
+const Iterations = ({ id, selectedIteration, addIteration, removeIteration }: IterationsParams): JSX.Element => {
+  const { apiToken } = parseCookies(); //todo: move to redux
```
Comment (Kiri23) [inline components/Iterations.tsx:17]: +1 for the todo comment 👏

File: components/Iterations.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      const getIterations = async () => {
+        const result = await fetch(
+          `https://www.pivotaltracker.com/services/v5/projects/${id}/iterations?limit=20&scope=current_backlog`,
+          {
+            headers: {
+              'X-TrackerToken': apiToken,
+            },
+          }
+        );
+
```
Comment (Kiri23) [inline components/Iterations.tsx:36]: What do you mean by moving this out to an action? are you referring to the network request

File: components/Iterations.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  const formatDate = (date: string): string => format(parseISO(date), 'do MMM yy');
+
+  return (
+    <ButtonDropdown>
+      <ButtonDropdown.Item
+        main={!Boolean(selectedIteration)} //todo: add the selected iteration to state.
+        onClick={(): void => {
+          //todo: logic for add/remove
+          removeIteration('iterations', null);
```
Comment (Kiri23) [inline components/Iterations.tsx:52]: isn't this completed yet? when I click on the `'No iteration` item the currently selected interaction is removed.

File: pages/projects/[id].tsx (hunk 1 above)
Comment (mattvv) [reply pages/projects/[id].tsx:34]: Yeah I think that just defers the work until I need to use the iteration :(

File: components/Iterations.tsx (hunk 3 above)
Comment (mattvv) [reply components/Iterations.tsx:36]: Yeah basically, I don't think the network requests should belong inside the components, I need to refactor them somewhere else in the action model

File: components/Iterations.tsx (hunk 4 above)
Comment (mattvv) [reply components/Iterations.tsx:52]: You are correct sir!

File: pages/projects/[id].tsx (hunk 1 above)
Comment (jansen0825) [reply pages/projects/[id].tsx:34]: @diegofigs I think this can be fixed with the type guard trick you just thought me right?

**Pull request merged**